pkg/obsidian/obsidian.go

```diff
@@ -56,8 +51,42 @@ func WriteToObsidianVault(formattedExpenses []string, outputFilePath string) err
 
 	defer file.Close()
 
-	// TODO Appends should not write expenses that are already present. Will need to seek to figure this out.
-	for _, formattedExpense := range formattedExpenses {
+	fileInfo, err := os.Stat(outputFilePath)
+	if err != nil {
+		return err
+	}
+
+	// Every expense will definitely be in the future from this date.
+	lastWrittenExpenseDate := "2023-01-01"
+
+	// If expenses are already present in this file, find the last written expense.
+	if fileSize := fileInfo.Size(); fileSize > 0 {
+		lastWrittenExpense, err := findLastExpenseInFile(file, fileSize)
+
+		if err != nil {
+			return err
+		}
+
+		fmt.Println("Last already written expense is: ", lastWrittenExpense)
+
+		lastWrittenExpenseDate, err = getDateFromFormattedExpense(lastWrittenExpense)
+
+		if err != nil {
+			lastWrittenExpenseDate = "2023-01-01"
+			fmt.Printf("Unable to parse date from last line: \"%s\". Using default date \"%s\" instead",
+				lastWrittenExpense, lastWrittenExpenseDate)
+		}
```

Comment:
The function should return the moment an error occurs, the error occurence is not causing a return here.

Reply:
This is an error I am fine with suppressing.

Reply:
An error should be treated as such - an error. Package internal logic should ideally not depend on an error occurring. Package logic should be as simple as possible, and logic dictating such flow should happen in the application code. This is considered idiomatic in Go.

The program above could be restructured such that `package obsidian` returns the error when a parse failure happens, and the caller of the public API exported by `package obsidian` decides to retry with another flow that is provided by `package obsidian`.
If `package obsidian` is imported as a library by a third party, this is behavior they are inevitably locked into unless they vendor the library and maintain their own version of it. However, if this behavior is considered crucial to what you want the package to be, go ahead and keep this as is.